### PR TITLE
Modified command line tool for testing

### PR DESCRIPTION
**Pull request by kianjones9**
Signed-off-by: Kian Jones <kianjones@fb.com>

## Summary

Added purging functionality to subscriber_cli.py cmd line tool. Purging doesn't just delete the subscriber from the DB, but also triggers the to detach a particular IMSI.
 
## Test Plan

PR will eventually be updated with corresponding tests

### CODE REVIEW

**Review by pshelar — COMMENTED** [review]
how is this functionally different from `s6a_service_cli` delete command?
 https://github.com/magma/magma/blob/f8b7931be64cbf55c50d55f9fca3a7c6a958bfb2/lte/gateway/python/scripts/s6a_service_cli.py#L43

**Comment by kianjones9** [comment]
Wow! I didn't know that existed! I want to make sure this serves my purpose before I close this PR. When I tried to use it I got an error:

```
magma@handset-test:~$ s6a_service_cli.py delete IMSI001010000091009 IMSI001010000091113 IMSI001010000091114 IMSI001010000051021 IMSI001010000051024 IMSI001010000051029 IMSI310260648765492
deleting subs: ['IMSI001010000091009', 'IMSI001010000091113', 'IMSI001010000091114', 'IMSI001010000051021', 'IMSI001010000051024', 'IMSI001010000051029', 'IMSI310260648765492']
Error! [StatusCode.UNAVAILABLE] failed to connect to all addresses
```

**Comment by kianjones9** [comment]
Got it working. Assessing whether it fits my needs or not.

**Comment by ulaskozat** [comment]
Hey I thought the idea was to trigger "detach" for mme. Deleting a sub from the db does not trigger this. cc @pshelar @kianjones9

**Comment by kianjones9** [comment]
Yes I came to that conclusion with @sudhikan. I was able to use the offload request to force a TAU, demanding the handsets reattach. i.e. I decided to continue with that, rather than add the feature like we discussed. It is my intention to revisit this before the end of my internship if I have time, and to contribute the feature upstream.

**Comment by kianjones9** [comment]
cc @ulaskozat @pshelar

**Comment by ulaskozat** [comment]
This would be a good addition if you can close on this. cc @pshelar @sudhikan.

File: lte/gateway/python/magma/subscriberdb/rpc_servicer.py [hunk 1]
```diff
@@ -64,6 +66,28 @@ def DeleteSubscriber(self, request, context):
...
+    @return_void
+    def PurgeSubscriber(self, request, context):
+        """
+        Triggers a network initiated detach
+        """
+        self._print_grpc(request)
+        sid = SIDUtils.to_str(request)
+        logging.debug("Purge subscriber rpc for sid: %s", sid)
+        self._store.delete_subscriber(sid)
+
```
Comment (krramasw) [inline lte/gateway/python/magma/subscriberdb/rpc_servicer.py:78]: What if subsequent code fails. We shall not delete from subsciption db if rest of the logic fails.

File: lte/gateway/python/scripts/subscriber_cli.py [hunk 2]
```diff
@@ -182,6 +187,7 @@ def create_parser():
     subparsers = parser.add_subparsers(title="subcommands", dest="cmd")
     parser_add = subparsers.add_parser("add", help="Add a new subscriber")
     parser_del = subparsers.add_parser("delete", help="Delete a subscriber")
+    parser_purge = subparsers.add_parser("purge", help="Purge a subscriber")
```
Comment (krramasw) [inline lte/gateway/python/scripts/subscriber_cli.py:190]: What is the difference between "delete subscriber" and "purge subscriber" ? Maybe it is explained in some doc and apologize for my ignorance.

**Comment by ekowtaylor** [comment]
@sudhikan, can you help move this PR forward. cc. @ulaskozat , @pshelar

**Comment by kianjones9** [comment]
I will close and revisit if/when I have time, but don't hold out for me! @ekowtaylor @sudhikan @pshelar @ulaskozat

**Pull request closed without merging**